flow/project.py

```diff
@@ -2520,28 +2694,37 @@ def _get_submission_operations(self, jobs, default_directives, names=None,
                                    ignore_conditions=IgnoreConditions.NONE,
                                    ignore_conditions_on_execution=IgnoreConditions.NONE):
         """Grabs _JobOperations that are eligible to run from FlowGroups."""
-        for job in jobs:
-            for group in self._gather_flow_groups(names):
-                if group.eligible(job, ignore_conditions) and self._eligible_for_submission(group,
-                                                                                            job):
+        for group in self._gather_flow_groups(names):
+            for job in jobs:
+                if (
+                    group._eligible([job], ignore_conditions) and
+                    self._eligible_for_submission(group, [job])
+                ):
                     yield group._create_submission_job_operation(
                         entrypoint=self._entrypoint,
                         default_directives=default_directives,
-                        job=job, index=0,
+                        jobs=[job], index=0,
                         ignore_conditions_on_execution=ignore_conditions_on_execution)
 
     def _get_pending_operations(self, jobs, operation_names=None,
                                 ignore_conditions=IgnoreConditions.NONE):
         "Get all pending operations for the given selection."
         assert not isinstance(operation_names, str)
-        for op in self._next_operations(* jobs, ignore_conditions=ignore_conditions):
+        for op in self._next_operations(jobs, ignore_conditions):
             if operation_names is None or any(re.fullmatch(n, op.name) for n in operation_names):
                 yield op
 
     def _verify_group_compatibility(self, groups):
         """Verifies that all selected groups can be submitted together."""
         return all(a.isdisjoint(b) for a in groups for b in groups if a != b)
 
+    def _verify_aggregate_project(self, aggregate):
```

Comment:
Given the internal nature of `_JobOperation` now this is likely not necessary once `JobOperation` is deprecated completely.

I would make a developer comment regarding that for the future.

Reply:
@b-butler  I'm not sure I completely follow what you suggest here.
Can you please help me understand?

Reply:
Actually, I am wrong on the point above. However, I would look for where users supply jobs in the API, (e.g. `FlowProject.submit` and such) and place the verification there. I don't think we would ever be in an internal state where the jobs we gather internally don't belong to the same project.